hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/AbstractAbfsTestWithTimeout.java

```diff
@@ -67,4 +77,42 @@ public void nameThread() {
   protected int getTestTimeoutMillis() {
     return TEST_TIMEOUT;
   }
+
+  /**
+   * Describe a test in the logs.
+   *
+   * @param text text to print
+   * @param args arguments to format in the printing
+   */
+  protected void describe(String text, Object... args) {
+    LOG.info("\n\n{}: {}\n",
+        methodName.getMethodName(),
+        String.format(text, args));
+  }
+
+  /**
+   * Validate Contents written on a file in Abfs
+   *
+   * @param fs AzureBlobFileSystem
+   * @param path Path of the file
+   * @param originalByteArray original byte array
+   * @return
+   * @throws IOException
+   */
+  protected boolean validateContent(AzureBlobFileSystem fs, Path path,
+      byte[] originalByteArray)
+      throws IOException {
+    FSDataInputStream in = fs.open(path);
+    byte[] contentByteArray = new byte[originalByteArray.length];
+    int seekPos = 0;
+    while (in.read() != -1) {
```

Comment:
read() always moves the stream forward, so you don't need these seek/seekPos++; they can only slow things down

Reply:
I will still need seekPos for indexing of contentByteArray..
seek() is however an extra method call, I can compare byte by byte with originalByteArray ?
Something like this ,maybe ?

` while (valueOfContentAtPos != -1 && pos < lenOfOriginalByteArray) {
      if (originalByteArray[pos] != valueOfContentAtPos)
        return false;
      valueOfContentAtPos = (byte) in.read();
      pos++;
    }
    if (valueOfContentAtPos != -1)
      return false;
    return true;`

Reply:
Yes this will be a better approach as it will take less time and space. Also as Steve suggested you don't need seek() here as read() maintains its own pointer. Also seek() is a very slow operation.